pocs/camera/sbig.py

```diff
@@ -221,91 +138,35 @@ def take_exposure(self,
-        # Add FITS headers from info the same as images.cr2_to_fits()
-        self.logger.debug("Updating FITS headers: {}".format(file_path))
-        with fits.open(file_path, 'update') as f:
-            hdu = f[0]
-            hdu.header.set('IMAGEID', info.get('image_id', ''))
-            hdu.header.set('SEQID', info.get('sequence_id', ''))
-            hdu.header.set('FIELD', info.get('field_name', ''))
-            hdu.header.set('RA-MNT', info.get('ra_mnt', ''), 'Degrees')
-            hdu.header.set('HA-MNT', info.get('ha_mnt', ''), 'Degrees')
-            hdu.header.set('DEC-MNT', info.get('dec_mnt', ''), 'Degrees')
-            hdu.header.set('EQUINOX', info.get('equinox', 2000.))  # Assume J2000
-            hdu.header.set('AIRMASS', info.get('airmass', ''), 'Sec(z)')
-            hdu.header.set('FILTER', info.get('filter', ''))
-            hdu.header.set('LAT-OBS', info.get('latitude', ''), 'Degrees')
-            hdu.header.set('LONG-OBS', info.get('longitude', ''), 'Degrees')
-            hdu.header.set('ELEV-OBS', info.get('elevation', ''), 'Meters')
-            hdu.header.set('MOONSEP', info.get('moon_separation', ''), 'Degrees')
-            hdu.header.set('MOONFRAC', info.get('moon_fraction', ''))
-            hdu.header.set('CREATOR', info.get('creator', ''), 'POCS Software version')
-            hdu.header.set('INSTRUME', info.get('camera_uid', ''), 'Camera ID')
-            hdu.header.set('OBSERVER', info.get('observer', ''), 'PANOPTES Unit ID')
-            hdu.header.set('ORIGIN', info.get('origin', ''))
-            hdu.header.set('RA-RATE', info.get('tracking_rate_ra', ''), 'RA Tracking Rate')
-
-        if info['is_primary']:
-            self.logger.debug("Extracting pretty image")
-            images.make_pretty_image(file_path, title=info['field_name'], primary=True)
-
-            self.logger.debug("Adding current observation to db: {}".format(image_id))
-            self.db.insert_current('observations', info, include_collection=False)
-        else:
-            self.logger.debug('Compressing {}'.format(file_path))
-            images.fpack(file_path)
-
-        self.logger.debug("Adding image metadata to db: {}".format(image_id))
-        self.db.observations.insert_one({
-            'data': info,
-            'date': current_time(datetime=True),
-            'type': 'observations',
-            'image_id': image_id,
-        })
-
-        # Mark the event as done
-        signal_event.set()
+        # Unbinned. Need to chance if binning gets implemented.
+        readout_mode = 'RM_1X1'
+
+        header.set('CAM-FW', self._info['firmware version'], 'Camera firmware version')
+        header.set('XPIXSZ', self._info['readout modes'][readout_mode]['pixel width'].value,
+                   'Microns')
+        header.set('YPIXSZ', self._info['readout modes'][readout_mode]['pixel height'].value,
+                   'Microns')
+        header.set('EGAIN', self._info['readout modes'][readout_mode]['gain'].value,
```

Comment:
I want to say we have a `GAIN` that comes of the Canon DSLR. Is `EGAIN` a better name?

Reply:
`EGAIN` (electrical gain? electron gain? electronic gain?) is what's used in SBIG's (unofficial) extension to the FITS standard (http://diffractionlimited.com/sbig-fits-extensions-sbigfitsext-1-0/) so that's what I used here.

Reply:
I was just wondering. I'll double-check on what goes in. We're also not reading it back out for anything at the moment.